src/gcp/secretManager.ts

```diff
@@ -63,123 +95,146 @@ export async function secretExists(projectId: string, name: string): Promise<boo
 }
 
 export function parseSecretResourceName(resourceName: string): Secret {
-  const nameTokens = resourceName.split("/");
+  const tokens = resourceName.match(SECRET_NAME_REGEX);
+  if (tokens == null) {
+    throw new FirebaseError(`Invalid secret resource name [${resourceName}].`);
+  }
   return {
-    projectId: nameTokens[1],
-    name: nameTokens[3],
+    projectId: tokens[1],
+    name: tokens[2],
   };
 }
 
 export function parseSecretVersionResourceName(resourceName: string): SecretVersion {
-  const nameTokens = resourceName.split("/");
+  const tokens = resourceName.match(SECRET_VERSION_NAME_REGEX);
+  if (tokens == null) {
+    throw new FirebaseError(`Invalid secret version resource name [${resourceName}].`);
+  }
   return {
     secret: {
-      projectId: nameTokens[1],
-      name: nameTokens[3],
+      projectId: tokens[1],
+      name: tokens[2],
     },
-    versionId: nameTokens[5],
+    version: tokens[3],
   };
 }
 
 export function toSecretVersionResourceName(secretVersion: SecretVersion): string {
-  return `projects/${secretVersion.secret.projectId}/secrets/${secretVersion.secret.name}/versions/${secretVersion.versionId}`;
+  return `projects/${secretVersion.secret.projectId}/secrets/${secretVersion.secret.name}/versions/${secretVersion.version}`;
```

Comment:
FWIW, Joe was technically correct to call it a "secretId" a URI component is an ID. If you don't include "id" you're talking about a "name" which is the full path.

Reply:
Well version is a component of a SecretVersion. I think more correct term for this would've been `secretVersoin.id`, but that looks awfully confusing to me.

Reply:
Oh yeah, versionId here and secretId elsewhere. You wouldn't need to fully namespace the variable name. just like it's not a projectsLocationFunctionId it's just a functionId.

Reply:
Fixed.